src/transport/PASESession.cpp

```diff
@@ -546,28 +546,25 @@ CHIP_ERROR PASESession::SendMsg1()
     uint8_t X[kMAX_Point_Length];
     size_t X_len = sizeof(X);
 
-    System::PacketBufferHandle msg_pA;
+    ReturnErrorOnFailure(mSpake2p.BeginProver(reinterpret_cast<const uint8_t *>(""), 0, reinterpret_cast<const uint8_t *>(""), 0,
+                                              &mPASEVerifier[0][0], kSpake2p_WS_Length, &mPASEVerifier[1][0], kSpake2p_WS_Length));
 
-    CHIP_ERROR err = mSpake2p.BeginProver(reinterpret_cast<const uint8_t *>(""), 0, reinterpret_cast<const uint8_t *>(""), 0,
-                                          &mPASEVerifier[0][0], kSpake2p_WS_Length, &mPASEVerifier[1][0], kSpake2p_WS_Length);
-    SuccessOrExit(err);
-
-    err = mSpake2p.ComputeRoundOne(X, &X_len);
-    SuccessOrExit(err);
+    ReturnErrorOnFailure(mSpake2p.ComputeRoundOne(X, &X_len));
 
-    msg_pA = System::PacketBufferHandle::NewWithData(&X[0], X_len);
-    VerifyOrExit(!msg_pA.IsNull(), err = CHIP_SYSTEM_ERROR_NO_MEMORY);
+    Encoding::LittleEndian::PacketBufferWriter bbuf(System::PacketBufferHandle::New(sizeof(uint16_t) + X_len));
+    VerifyOrReturnError(!bbuf.IsNull(), CHIP_SYSTEM_ERROR_NO_MEMORY);
+    bbuf.Put16(mConnectionState.GetLocalKeyID());
+    bbuf.Put(&X[0], X_len);
```

Comment:
what is the size of this Put?

since X_len is size_t, I wonder if this will generate different buffers between 32bit and 64bit systems. Does it matter?

I would much rather have Put8/Put16/Put32/Put64 in code to ensure it is consistent

Reply:
@andy31415, `X_len` is used to indicate the size of `&X[0]` buffer. The value of `X_len` itself is not getting written in the output buffer, or sent over the wire as a 32bit/64bit value.

Reply:
ah ... need more caffeine when reading PRs.